src/extensions/scratch3_vizblocks/index.js

```diff
@@ -136,6 +155,268 @@ class Scratch3VizBlocks {
+                        default: 'draw axis-Y',
+                        description: 'draw axis-Y'
+                    })
+                },
+                {
+                    opcode: 'readXY',
+                    blockType: BlockType.COMMAND,
+                    text: formatMessage({
+                        id: 'vizblocks.read',
+                        default: 'read data x:[X] y:[Y]',
+                        description: 'read data from (X, Y)'
+                    }),
+                    arguments: {
+                        X: {
+                            type: ArgumentType.NUMBER,
+                            defaultValue: 0
+                        },
+                        Y: {
+                            type: ArgumentType.NUMBER,
+                            defaultValue: 0
+                        }
+                    }
+                },
+                {
+                    opcode: 'drawLine',
+                    blockType: BlockType.COMMAND,
+                    text: formatMessage({
+                        id: 'vizblocks.drawLine',
+                        default: 'draw line',
+                        description: 'draw line'
+                    })
+                },
+                {
+                    opcode: 'clear',
+                    blockType: BlockType.COMMAND,
+                    text: formatMessage({
+                        id: 'vizblocks.clear',
+                        default: 'clear',
+                        description: 'clear canvas'
+                    })
+                }
+            ]
+        };
+    }
+
+    /**
+     * The pen "clear" block clears the pen layer's contents.
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    clear (args, util) {
+        const penSkinId = this._getPenLayerID();
+        const target = util.target;
+        if (penSkinId >= 0) {
+            this.runtime.renderer.penClear(penSkinId);
+            this.runtime.requestRedraw();
+        }
+
+        if (!this._posEmpty){
+            this._xPos.length = 0;
+            this._yPos.length = 0;
+            this._posEmpty = true;
+        }
+
+        target.sprite.costumes_.forEach(costume => {
+            if (costume.name !== 'costume1' && costume.name !== 'costume2') {
+                // eslint-disable-next-line no-console
+                console.log(costume.name);
+                target.deleteCostume(target.getCostumeIndexByName(costume.name));
+            }
+        });
+
+    }
+
+    /**
+     * Draw X axis using functions from "pen block"
+     * @param {object} args - the block arguments.
+     * @param {object} util - utility object provided by the runtime.
+     */
+    drawAxisX (args, util) {
+        const target = util.target;
+        
+        // console.log('draw axis');
+        target.setVisible(false);
+        this._costumes = this.loadCostumes(util);
```

Comment:
We can pass `target` into the function directly.

```suggestion
        this._costumes = this.loadCostumes(target);
```